File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Membership.ChangePasswordForm.ts
namespace Serene.Membership {
    export class ChangePasswordForm extends Serenity.PrefixedContext {
        static formKey = 'Membership.ChangePassword';

    }

    export interface ChangePasswordForm extends Serenity.PrefixedContext {
        OldPassword(): Serenity.PasswordEditor;
        NewPassword(): Serenity.PasswordEditor;
        ConfirmPassword(): Serenity.PasswordEditor;
    }

    [['OldPassword', Serenity.PasswordEditor], ['NewPassword', Serenity.PasswordEditor], ['ConfirmPassword', Serenity.PasswordEditor]].forEach(x => ChangePasswordForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

File: Serene/Serene.Script/Imports/ServerImports/Membership.ChangePasswordForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Membership
{
    public partial class ChangePasswordForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Membership.ChangePassword";

        public ChangePasswordForm(string idPrefix) : base(idPrefix) {}

        public PasswordEditor OldPassword { [InlineCode("{this}.w('OldPassword', Serenity.PasswordEditor)")] get; private set; }
        public PasswordEditor NewPassword { [InlineCode("{this}.w('NewPassword', Serenity.PasswordEditor)")] get; private set; }
        public PasswordEditor ConfirmPassword { [InlineCode("{this}.w('ConfirmPassword', Serenity.PasswordEditor)")] get; private set; }
    }
}

